staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/status.go

```diff
@@ -58,11 +60,12 @@ func ErrorToAPIStatus(err error) *metav1.Status {
 		//TODO: check for invalid responses
 		return &status
 	default:
-		status := http.StatusInternalServerError
 		switch {
-		//TODO: replace me with NewConflictErr
 		case storage.IsConflict(err):
-			status = http.StatusConflict
+			status := &errors.NewConflict(schema.GroupResource{}, "", err).ErrStatus
+			status.Kind = "Status"
+			status.APIVersion = "v1"
+			return status
```

Comment:
As a side-effect of this return the error will not be logged. Is it expected? We will only skip logging of the conflict error here, which seems questionable. Did you consider to setup "reason" and "message" fields at the beginning of the `default` branch (like for status) and update specifically for conflict errors?